csharp/src/SeedLang/Visualization/Value.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using SeedLang.Runtime;
+
+namespace SeedLang.Visualization {
+  // A wrapper class of the underlying VMValue. It's used to represent runtime values in the
+  // notification events of visualizers.
+  public class Value : IEquatable<Value> {
+    public bool IsNil => _value.IsNil;
+
+    public bool IsBoolean => _value.IsBoolean;
+    public bool IsNumber => _value.IsNumber;
+    public bool IsString => _value.IsString;
+
+    public bool IsDict => _value.IsDict;
+    public bool IsList => _value.IsList;
+    public bool IsTuple => _value.IsTuple;
+
+    public int Length => _value.Length;
+
+    private readonly VMValue _value;
+
+    public Value() {
```

Comment:
Do we really need these public constructors for visualizers to instantiate a Value? What are the relevant use cases?

```
    public Value() {
      _value = new VMValue();
    }

    public Value(bool boolean) {
      _value = new VMValue(boolean);
    }

    public Value(double number) {
      _value = new VMValue(number);
    }

    public Value(string str) {
      _value = new VMValue(str);
    }
```

A potential use case could be that a visualizer has the ability to update the value of a variable during runtime. It is a complex case. I think we can probably postpone this "live-update" feature to some time later.

I think in most common cases, visualizers simply read the value of a runtime value, and do not need constructors of the Value class.

Reply:
One scenario is used as a key or index for the subscript operation. For example, if the user get a VTag notification like `Swap(array, i, j)` or `Access(dict, key)`, he can use `array[i]` or `dict[key]`. But he has to create a value if he world like to access other elements, like:
```python
array[new Value(1)]
dict[new Value("name")]
```
@wixette Let's decide if this use case need be support now.

Reply:
Ah, you're right. We need those constructors to support index-access and key-access.